File: src/types/payment.ts
/**
 * Payment Types - Shared across all portals
 */

export type PaymentMethod = 'online' | 'cod';

export type PaymentStatus = 
  | 'pending' 
  | 'processing' 
  | 'success' 
  | 'failed' 
  | 'refunded';

export type RefundStatus = 
  | 'pending' 
  | 'processing' 
  | 'completed' 
  | 'failed';

export type SettlementStatus = 
  | 'pending' 
  | 'processing' 
  | 'completed' 
  | 'failed' 
  | 'on_hold';

export interface Payment {
  id: string;
  orderId: string;
  buyerId: string;
  vendorId?: string | null;
  amount: number;
  currency: string;
  paymentMethod: PaymentMethod;
  paymentStatus: PaymentStatus;
  razorpayOrderId?: string | null;
  razorpayPaymentId?: string | null;
  paymentGateway?: string | null;
  transactionId?: string | null;
  paymentErrorCode?: string | null;
  paymentErrorMessage?: string | null;
  metadata?: string | null;
  notes?: string | null;
  createdAt: string;
  updatedAt: string;
  completedAt?: string | null;
  failedAt?: string | null;
}

export interface CreatePaymentRequest {
  orderId: string;
  buyerId: string;
  vendorId?: string;
  amount: number;
  currency?: string;
  paymentMethod: PaymentMethod;
  metadata?: string;
  notes?: string;
}

export interface VerifyPaymentRequest {
  razorpayOrderId: string;
  razorpayPaymentId: string;
  razorpaySignature: string;
}

export interface Refund {
  id: string;
  paymentId: string;
  amount: number;
  currency: string;
  refundReason: string;
  refundStatus: RefundStatus;
  razorpayRefundId?: string | null;
  refundErrorCode?: string | null;
  refundErrorMessage?: string | null;
  initiatedBy?: string | null;
  notes?: string | null;
  createdAt: string;
  updatedAt: string;
  completedAt?: string | null;
}

export interface CreateRefundRequest {
  paymentId: string;
  amount?: number; // Full refund if not specified
  reason: string;
  notes?: string;
}

export interface Settlement {
  id: string;
  vendorId: string;
  vendorName: string;
  totalAmount: number;
  commissionPercentage: number;
  commissionAmount: number;
  taxAmount: number;
  adjustmentAmount: number;
  settlementAmount: number;
  paymentCount: number;
  periodStart: string;
  periodEnd: string;
  status: SettlementStatus;
  accountNumber?: string | null;
  ifscCode?: string | null;
  accountHolderName?: string | null;
  utrNumber?: string | null;
  settlementMethod?: string | null;
  notes?: string | null;
  createdAt: string;
  updatedAt: string;
  processedAt?: string | null;
  payments?: Payment[]; // Included payments in this settlement
}

export interface GenerateSettlementRequest {
  vendorId: string;
  periodStart: string;
  periodEnd: string;
  commissionPercentage?: number;
  adjustmentAmount?: number;
  notes?: string;
}

export interface SettlementLineItem {
  id: string;
  settlementId: string;
  paymentId: string;
  orderId: string;
  amount: number;
  commissionAmount: number;
  createdAt: string;
}

export interface ApiResponse<T> {
  success: boolean;
  data?: T;
  error?: {
    code: string;
    message: string;
    details?: string;
  };
  timestamp?: string;
}

export interface PaginatedResponse<T> {
  items: T[];
  pagination: {
    page: number;
    pageSize: number;
    totalItems: number;
    totalPages: number;
    hasNextPage: boolean;
    hasPreviousPage: boolean;
  };
}

export interface PaymentFilters {
  status?: PaymentStatus | 'all';
  paymentMethod?: PaymentMethod | 'all';
  dateFrom?: string;
  dateTo?: string;
  search?: string;
}

export interface SettlementFilters {
  status?: SettlementStatus | 'all';
  vendorId?: string;
  dateFrom?: string;
  dateTo?: string;
  search?: string;
}

File: backend/src/services/PaymentService/Models/Entities/Payment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PaymentService.Models.Entities;

[Table("payments")]
public class Payment
{
    [Key]
    [Column("id")]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    [Column("order_id")]
    public Guid OrderId { get; set; }

    [Required]
    [Column("buyer_id")]
    public Guid BuyerId { get; set; }

    [Column("vendor_id")]
    public Guid? VendorId { get; set; }

    [Required]
    [Column("amount")]
    [Precision(18, 2)]
    public decimal Amount { get; set; }

    [Required]
    [Column("currency")]
    [StringLength(3)]
    public string Currency { get; set; } = "INR";

    [Required]
    [Column("payment_method")]
    [StringLength(50)]
    public string PaymentMethod { get; set; } = null!; // online, cod, upi, card, netbanking

    [Required]
    [Column("payment_status")]
    [StringLength(50)]
    public string PaymentStatus { get; set; } = "pending"; // pending, processing, success, failed, refunded

    // Razorpay fields
    [Column("razorpay_order_id")]
    [StringLength(100)]
    public string? RazorpayOrderId { get; set; }

    [Column("razorpay_payment_id")]
    [StringLength(100)]
    public string? RazorpayPaymentId { get; set; }

    [Column("razorpay_signature")]
    [StringLength(200)]
    public string? RazorpaySignature { get; set; }

    // Payment details
    [Column("payment_gateway")]
    [StringLength(50)]
    public string? PaymentGateway { get; set; } // razorpay, paytm, phonepe

    [Column("transaction_id")]
    [StringLength(100)]
    public string? TransactionId { get; set; }

    [Column("payment_error_code")]
    [StringLength(50)]
    public string? PaymentErrorCode { get; set; }

    [Column("payment_error_message")]
    [StringLength(500)]
    public string? PaymentErrorMessage { get; set; }

    // Metadata
    [Column("metadata")]
    [StringLength(2000)]
    public string? Metadata { get; set; } // JSON string

    [Column("notes")]
    [StringLength(1000)]
    public string? Notes { get; set; }

    // Timestamps
    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    [Column("completed_at")]
    public DateTime? CompletedAt { get; set; }

    [Column("failed_at")]
    public DateTime? FailedAt { get; set; }

    // Navigation properties
    public virtual ICollection<PaymentRefund> Refunds { get; set; } = new List<PaymentRefund>();
    public virtual ICollection<PaymentWebhook> Webhooks { get; set; } = new List<PaymentWebhook>();
}
